autoscaler/controllers/datacollection/daemonset.go

```diff
@@ -289,3 +294,38 @@ func patchDaemonSet(existing *appsv1.DaemonSet, desired *appsv1.DaemonSet, ctx c
 func boolPtr(b bool) *bool {
 	return &b
 }
+
+type DelayManager struct {
+	mu         sync.Mutex
+	inProgress bool
+}
+
+// runFunctionWithDelayAndSkipNewCalls runs the function with the specified delay and skips new calls until the function execution is finished
+func (dm *DelayManager) runFunctionWithDelayAndSkipNewCalls(delay time.Duration, fn func(args ...interface{}) (*appsv1.DaemonSet, error), fnArgs ...interface{}) {
+	dm.mu.Lock()
+	defer dm.mu.Unlock()
+	logger := log.FromContext(fnArgs[2].(context.Context))
+
+	if dm.inProgress {
+		logger.Info("Function execution in progress. Skipping...")
+		return
+	}
+
+	dm.inProgress = true
+
+	time.AfterFunc(delay, func() {
+		dm.mu.Lock()
+		defer dm.mu.Unlock()
+
+		logger.Info("Sync DaemonSet function execution started...")
```

Comment:
I can also create a generic Retry function / use external library like: https://github.com/avast/retry-go
Or leave it as is..
WDYT?